framework/src/engine/generator/network_endpoint.ts

```diff
@@ -231,14 +231,7 @@ export class NetworkEndpoint extends BaseNetworkEndpoint {
 					await this._receiveTransaction(transaction);
 				}
 			} catch (err) {
-				// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
 				this._logger.warn({ err, peerId }, 'Received invalid transactions.');
-				if (err instanceof InvalidTransactionError) {
```

Comment:
I think
```
transaction.validate()
```
is missing, and if it fails on `fromBytes` or `validate`, it should apply penalty to the peer